bin/replicate-counts.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+"""
+This script should take in a dataset, sum the raw counts 
+from all replicate sequences in the library, then proceed to
+set the value for each replicate to that sum
+
+Currently, this function only sets the raw counts, in place.
+"""
+
+import pandas as pd
+import numpy as np
+import phippery
+from phippery.phipdata import get_annotation_table
+import sys
+
+ds = phippery.load(sys.argv[1])
+
+# find all value counts greater than 1,
+pep_anno_table = get_annotation_table(ds, "peptide")
+oligo_vc = pep_anno_table["Oligo"].value_counts()
+
+# for each oligo that is not unique in a library
+for oligo, count in oligo_vc[oligo_vc > 1].items():
```

Comment:
This pattern may be a little easier using something like:

```
# Iterate over every group of peptides which share the same oligo sequence
for oligo_seq, pep_anno_table_oligo in pep_anno_table.groupby('Oligo'):

    # Check to see if there are multiple peptides with the same oligo sequence
    if pep_anno_table_oligo.shape[0] == 1:

        # Don't make any changes for unique oligos
        continue

    # Otherwise, get the sum of the counts across all oligos
    rep_pep_sums = pep_anno_table_oligo.sum(axis=0).values

    # Set the summed value for all peptides which share the same oligo sequence
    ds.counts.loc[pep_anno_table_oligo.index.values, :] = np.tile(rep_pep_sums, (pep_anno_table_oligo.shape[0], 1))
```

Reply:
Nice! yea I like it!